codex/codex.nim

```diff
@@ -130,6 +130,7 @@ proc new*(T: type CodexServer, config: CodexConf, privateKey: CodexPrivateKey):
 
   if config.cacheSize > 0:
     cache = CacheStore.new(cacheSize = config.cacheSize * MiB)
+    ## Is unused?
```

Comment:
If you can explain the plan for this cacheStore, I'll happily take a stab at it.

Reply:
The cache store is just an LRU in memory store, which we use as a cache. It was introduced before we had the `Datastore` abstraction, so it might be a little out of place right now. We can still be used but we might just want to rewrite it as a `Datastore`, which makes it more generic. 

On the other hand, the block store abstraction is just as good, so it might make sense to continue using as is - I'm not sure, certainly the `Datastore` route is more work because we don't have it implemented.

In any case, this would required either explicit calls to the cache, before falling back to the network or local store, or use something like a `TieredDatastore` to dispatch the calls for us.

Reply:
I understand what the CacheStore does. I don't understand why it does it. Can you explain to me a use-case where this would be used?

I think it would be pretty nice to implement a tiered-date-store module: When you call it, it'll pass the call to any number of other datastores in order. In practice that could be cache, local, network. Although, I imagine this makes more sense of getting blocks than it does for putting blocks.

Either way, this should probably be a new branch and a new PR, I'd say. Let's wrap this one up first, yeah? :D

Reply:
The `CacheStore` stores unserialized blocks in memory, it helps when the block needs to be read from a slow medium, like a spinning drive. You probably won't notice it's absence with a fast SSD tho, thats why it should be optional.

Reply:
> Either way, this should probably be a new branch and a new PR, I'd say. Let's wrap this one up first, yeah? :D

Totally, it needs its own PR.